### ISSUE TITLE
Remove lodash dependency

### ISSUE THREAD

**Issue by jwerre** [node-oauth/node-oauth2-server#23]
Just did a quick search for "_." and realized `lodash` is a dependency almost exclusively because of [`assign`](https://lodash.com/docs/4.17.15#assign). We could easily remove `lodash` (and all the security concerns with it) by using the Spread operator or `Object.assign`. 

[`_.includes`](https://lodash.com/docs/4.17.15#includes) and [`_.has`](https://lodash.com/docs/4.17.15#has) are also used a couple of times and could easily be replaced with [`Array.included`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) and [`hasOwnProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty).

**Comment by jankapunkt** [comment]
Yes this is very good but let's be very cautious and add a few extra tests, if required in order to make sure we don't break things.
